### PR TITLE
Added Api Calls for Ref Archs (#KSriHarsha)

### PR DESCRIPTION
**Pull request by raxkumar-wedaa**
- Added Api Calls for Ref Archs

### CODE REVIEW

**Review by raxkumar-wedaa — COMMENTED** [review]
Review OK!
(#rakesh)

File: model/refArchitecture.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+var refArchSchema = new Schema({
+    name :{
+        type: String,
+        unique : false,
```
Comment (rohith-wedaa) [inline model/refArchitecture.js]: Name must also be unique. There cannot be more than 1 reference architecture having **Full Stack** name, which causes ambiguity.

File: model/refArchitecture.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        required : true
+    },
+    refArch_id :{
```
Comment (rohith-wedaa) [inline model/refArchitecture.js]: let's use mongo's `_id`.

File: model/refArchitecture.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        type: String,
+        unique : true,
+        required : true
+    },
```
Comment (rohith-wedaa) [inline model/refArchitecture.js:9]: let's add one more field called `type`. there can be reference architectures for application, infrastructure, data pipelines or devsecops.

**Review by rohith-wedaa — CHANGES_REQUESTED** [review]

**Review by rohith-wedaa — APPROVED** [review]

**Pull request merged**